src/ServiceControl/ExternalIntegrations/MessageFailureResolvedByRetryPublisher.cs

```diff
@@ -0,0 +1,35 @@
+namespace ServiceControl.ExternalIntegrations
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Contracts.MessageFailures;
+    using Raven.Client;
+
+    class MessageFailureResolvedByRetryPublisher : EventPublisher<MessageFailureResolvedByRetry, MessageFailureResolvedByRetryPublisher.DispatchContext>
+    {
+        protected override DispatchContext CreateDispatchRequest(MessageFailureResolvedByRetry @event)
+        {
+            return new DispatchContext
+            {
+                FailedMessageId = new Guid(@event.FailedMessageId)
+            };
+        }
+
+        protected override Task<IEnumerable<object>> PublishEvents(IEnumerable<DispatchContext> contexts, IAsyncDocumentSession session)
+        {
+            return Task.FromResult(contexts.Select(r => (object)new Contracts.MessageFailureResolvedByRetry
+            {
+                FailedMessageId = r.FailedMessageId.ToString(),
+                AlternativeFailedMessageIds = r.AlternativeFailedMessageIds
```

Comment:
Isn't this information always empty? We have `RetryHandlerForBackwardsCompatability` and it doesn't set the array. Ah we set it in the recoverability specific enricher in the audit pipeline. 

```
            IEnumerable<string> GetAlternativeUniqueMessageId(IReadOnlyDictionary<string, string> headers)
            {
                var messageId = headers.MessageId();
                if (headers.TryGetValue(Headers.ProcessingEndpoint, out var processingEndpoint))
                {
                    yield return DeterministicGuid.MakeId(messageId, processingEndpoint).ToString();
                }

                if (headers.TryGetValue("NServiceBus.FailedQ", out var failedQ))
                {
                    yield return DeterministicGuid.MakeId(messageId, ExtractQueueNameForLegacyReasons(failedQ)).ToString();
                }

                if (headers.TryGetValue(Headers.ReplyToAddress, out var replyToAddress))
                {
                    yield return DeterministicGuid.MakeId(messageId, ExtractQueueNameForLegacyReasons(replyToAddress)).ToString();
                }
            }
```

How is that information relevant for the end users? Can you elaborate what they would do with this information?

Reply:
In the case I was looking at the customer wants to use the events to reconcile his backup error collection, so he wants to know if he needs to remove a failed message from his backup. (so if it was resolved, archived or unarchived)